src/invidious/routes/channels.cr

```diff
@@ -1,6 +1,12 @@
 {% skip_file if flag?(:api_only) %}
 
 module Invidious::Routes::Channels
+  # Redirection for unsupported routes ("tabs")
+  def self.redirect_home(env)
+    ucid = env.params.url["ucid"]
+    return env.redirect "/channel/#{URI.encode_www_form(ucid)}"
```

Comment:
```suggestion
    return env.redirect "/channel/#{URI.encode_path_segment(ucid)}"
```

Reply:
This is on purpose, to prevent path transversal attempts. All segments in `env.params.url` are `URI.decode`'d, so if we use that value directly or with `encode_path_segment` (which doesn't encode `/`), a malicious URI like `/channel/UCxxxx%2F..%2F..%2Fsomething` would redirect to `/channel/UCxxxx/../../something`.

In theory, there is no vulnerable path (all critical routes like `/delete_account` will show a confirmation prompt on a GET request), but I prefer to be on the safe side.